### PR TITLE
Alerting: Load alerting state view based on prom primary feature toggle

### PR DESCRIPTION
**Pull request by tomratcliffe**
**What is this feature?**
Fixes rendering of alert rule state list view on initial load

**Why do we need this feature?**
Trying to link directly to the state view of the page wasn't working as intended.

**Who is this feature for?**

Alerting UI users

### CODE REVIEW

File: public/app/features/alerting/unified/components/rules/RulesTable.tsx [hunk 1]
```diff
@@ -146,14 +146,10 @@ function useLazyLoadRulerRules(rules: CombinedRule[]) {
...
-    } else {
-      // We need to reset the actions to update the rules if they changed
-      // Otherwise useAsync acts like a cache and always return the first rules passed to it
-      actions.reset();
     }
-  }, [rules, actions]);
+  }, [actions]);
 
-  return state;
+  return prometheusRulesPrimary ? state : { result: rules, status: 'success' };
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/rules/RulesTable.tsx:152]: How about wrappiing the second object into a `useMemo`?
Just to keep the reference the same between rerenders

**Review by gillesdemey — APPROVED** [review]
Code changes LGTM; no nits

File: public/app/features/alerting/unified/components/rules/RulesTable.tsx [hunk 2]
```diff
@@ -63,9 +63,19 @@ export const RulesTable = ({
...
   const { pageItems, page, numberOfPages, onPageChange } = usePagination(rules, 1, DEFAULT_PER_PAGE_PAGINATION);
 
-  const { result: rulesWithRulerDefinitions, status: rulerRulesLoadingStatus } = useLazyLoadRulerRules(pageItems);
+  const [lazyLoadRules, { result: rulesWithRulerDefinitions, status: rulerRulesLoadingStatus }] =
+    useLazyLoadRulerRules(pageItems);
+  const isLoadingRulerGroup = useMemo(
+    () =>
+      !rulerRulesLoadingStatus || rulerRulesLoadingStatus === 'loading' || rulerRulesLoadingStatus === 'not-executed',
+    [rulerRulesLoadingStatus]
+  );
```
Comment (tomratcliffe) [inline public/app/features/alerting/unified/components/rules/RulesTable.tsx:72]: Was attempting to make the loading work correctly but it flicks between the actions rendering partly, then more buttons, then as a skeleton, then finally back to rendering properly 🤷 Think something strange is going on (but this case only happens on first page load if you're going _straight_ to the state list view)

File: public/app/features/alerting/unified/components/rules/RulesTable.tsx [hunk 3]
```diff
@@ -63,9 +63,19 @@ export const RulesTable = ({
...
 
-  const isLoadingRulerGroup = rulerRulesLoadingStatus === 'loading';
+  useEffect(() => {
+    if (pageItems.length > 0) {
+      lazyLoadRules.execute();
+    }
+  }, [lazyLoadRules, pageItems, rulerRulesLoadingStatus]);
```
Comment (tomratcliffe) [inline public/app/features/alerting/unified/components/rules/RulesTable.tsx:78]: Moving the `execute` into here results in the state list rendering properly on initial page load. But there's a lot to unravel outside of this, and the logic/prop drilling that needs to happen to get to this point is quite complicated. 
I think overall, we're best revisiting/fixing this all once the new list view is closer to completion

**Review by konrad147 — APPROVED** [review]
LGTM!

**Review by konrad147 — APPROVED** [review]
LGTM!

**Pull request merged**